src/background/vmModules/ModuleManager.ts

```diff
@@ -27,7 +28,18 @@ const NAMESPACE_REGEX = new RegExp('^[-a-z0-9]{3,8}$');
 @singleton()
 export class ModuleManager {
   #_modules: Module[] | undefined;
-  #approvalController: ApprovalController;
+  #approvalController: BatchApprovalController;
+
+  isNonRestrictedMethod(module: Module, method: string): boolean {
+    const nonRestrictedMethods =
+      module.getManifest()?.permissions.rpc.nonRestrictedMethods;
+
+    if (nonRestrictedMethods === undefined) {
+      return false;
+    }
+
+    return nonRestrictedMethods.includes(method);
+  }
```

Comment:
Should this be a util instead since it's not using anything form the service state.